gui/editor/editor.cpp

```diff
@@ -1001,6 +1001,40 @@ void NumeReEditor::OnChar( wxStyledTextEvent& event )
     if (sAutoCompList.length())
         AutoCompShow(lenEntered, sAutoCompList);
 
+    // if line indicator setting is active, provide autowrapping for comments
+    bool bLineIndicatorSet = m_options->getSetting(SETTING_B_LINELENGTH).active();
+    const int iMaxLineLength = 100;     // member from codeanalyzer.hpp private MAXLINESOFCODE
+    const int iColumnPos = GetColumn(currentPos);
+
+    if (bLineIndicatorSet && (iColumnPos > iMaxLineLength))
+    {
+        int indentWidth = GetLineIndentation(currentLine);
+
+        if (GetStyleAt(currentPos) == wxSTC_NSCR_COMMENT_LINE)
+        {
+            InsertText(currentPos-1, "\r\n## ");

```

Comment:
Here (and all following occurences) the solution lacks a bit of cleverness: if I'm typing a long word, the solution will now break that word straight somewhere in the middle. You want to try using `WordStartPosition(currentPos)` as an alternative for calculating the best break position. The offset where you want to move the caret to is not dependent on this position. If that fails (it returns `wxNOT_FOUND` in this case), you do not want to break at all.

Reply:
Actually, you can make the condition even simpler like so:
```C++
int nBreakPos = WordStartPosition(currentPos);

// Wrap only, if the breaking point is more than 10 characters behind the indentation
if (nBreakPos > GetLineIndentPosition(GetCurrentLine()) + 10) // will automatically check for wxNOT_FOUND (==-1)
    // Perform wrapping ...
```

Reply:
what is number 10 here? is it the width of single character?

Reply:
No, that's some guess. You could also use a quarter of the maximal line length. The idea is that it doesn't make sense to wrap around, if the word to be wrapped around will probably be longer than the total number of characters (its breaking point is already close to the beginning of the current line).

Reply:
for this thing, I have made changes, but I'm still not sure if they are correct